src/core/common/string.cpp

```diff
@@ -66,13 +66,49 @@ const char *StringFind(const char *aString, char aChar)
     return ret;
 }
 
+const char *StringFind(const char *aString, const char *aSubString)
+{
+    const char *start = aString;
+    const char *sub   = aSubString;
+
+    while (true)
+    {
+        if (*sub == '\0')
+        {
+            break;
+        }
+
+        if (*aString == '\0')
+        {
+            start = nullptr;
+            break;
+        }
+
+        if (*aString++ != *sub++)
+        {
+            sub   = aSubString;
+            start = aString;
```

Comment:
Does this properly handle cases where the substring itself has repetition? For example:
* aString = "abcabcabcd"
* aSubString = "abcabcd"

Reply:
You are absolutely right! It does not. Thanks for catching this. 
Will fix it in next push.